great_expectations/execution_engine/polars_execution_engine.py

```diff
@@ -0,0 +1,640 @@
+            raise ValueError(
+                "Batch has not been loaded - please run load_batch_data() to load a batch."
+            )
+
+        return cast(PolarsBatchData, self.batch_manager.active_batch_data).dataframe
+
+    # NOTE Abe 20201105: Any reason this shouldn't be a private method?
+    @staticmethod
+    def guess_reader_method_from_path(path):
+        """Helper method for deciding which reader to use to read in a certain path.
+
+        Args:
+            path (str): the to use to guess
+
+        Returns:
+            ReaderMethod to use for the filepath
+
+        """
+        if path.endswith(".csv") or path.endswith(".tsv"):
+            return {"reader_method": "read_csv"}
+        elif (
+            path.endswith(".parquet") or path.endswith(".parq") or path.endswith(".pqt")
+        ):
+            return {"reader_method": "read_parquet"}
+        elif path.endswith(".xlsx") or path.endswith(".xls"):
+            return {"reader_method": "read_excel"}
+        elif path.endswith(".json"):
+            return {"reader_method": "read_json"}
+        elif path.endswith(".pkl"):
+            return {"reader_method": "read_pickle"}
+        elif path.endswith(".feather") or path.endswith(".ipc"):
+            return {"reader_method": "read_ipc"}
+        elif path.endswith(".avro"):
+            return {"reader_method": "read_avro"}
+
+        else:
+            raise ge_exceptions.ExecutionEngineError(
+                f'Unable to determine reader method from path: "{path}".'
+            )
+
+    def _get_reader_fn(self, reader_method=None, path=None):
+        """Static helper for parsing reader types. If reader_method is not provided, path will be used to guess the
+        correct reader_method.
+
+        Args:
+            reader_method (str): the name of the reader method to use, if available.
+            path (str): the path used to guess
+
+        Returns:
+            ReaderMethod to use for the filepath
+
+        """
+        if reader_method is None and path is None:
+            raise ge_exceptions.ExecutionEngineError(
+                "Unable to determine polars reader function without reader_method or path."
+            )
+
+        reader_options = {}
+        if reader_method is None:
+            path_guess = self.guess_reader_method_from_path(path)
+            reader_method = path_guess["reader_method"]
+            reader_options = path_guess.get(
+                "reader_options"
+            )  # This may not be there; use None in that case
+
+        try:
+            reader_fn = getattr(pl, reader_method)
+            if reader_options:
+                reader_fn = partial(reader_fn, **reader_options)
+            return reader_fn
+        except AttributeError:
+            raise ge_exceptions.ExecutionEngineError(
+                f'Unable to find reader_method "{reader_method}" in polars.'
+            )
+
+    def resolve_metric_bundle(
```

Comment:
Polars also supports [lazy evaluation](https://pola-rs.github.io/polars-book/user-guide/quickstart/intro.html#lazy-quick-start), maybe we can leverage that in a similar way to our spark execution engine? Might be out of the scope of this effort.